### PR TITLE
add hande

### PR DESCRIPTION
**Pull request by ltalirz**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    - cmake >=3.6
+    - make
+  host:
+    - {{ mpi }}  # [mpi != 'nompi']
+    - scalapack  # [mpi != 'nompi']
+    - libblas  # [osx]
+    - openblas-ilp64  # [linux]
+    # - mkl  # [linux]
+    - liblapack
+    - llvm-openmp  # [osx]
```
Comment (xhochy) [inline recipes/recipe/meta.yaml:43]: This should also be in build

File: recipes/recipe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+{% set name = "hande" %}
+{% set version = "1.5" %}
+
+# Define build matrix for MPI vs. non-mpi
+# ensure mpi is defined (needed for conda-smithy recipe-lint)
+{% set mpi = mpi or 'nompi' %}
+{% set build = 1 %}
+{% if mpi == 'nompi' %}
+# prioritize 'nompi' variant via build number
```
Comment (xhochy) [inline recipes/recipe/meta.yaml:9]: For priotization rather use track_festures, i.e. add a feature to the lower priority one.

File: recipes/recipe/build.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -ex
+
+
+if [[ "$mpi" == "nompi" ]]; then
+
+cmake -H. -Bbuild \
+    -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+    -DENABLE_OPENMP=ON \
+    -DHDF5_hdf5_fortran_LIBRARY=$PREFIX/lib/libhdf5.dylib
```
Comment (xhochy) [inline recipes/recipe/build.sh:10]:
```suggestion
    -DHDF5_hdf5_fortran_LIBRARY=$PREFIX/lib/libhdf5${SHLIB_EXT}
```

File: recipes/recipe/build.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -ex
+
+
+if [[ "$mpi" == "nompi" ]]; then
+
+cmake -H. -Bbuild \
```
Comment (xhochy) [inline recipes/recipe/build.sh:7]:
```suggestion
cmake ${CMAKE_ARGS} -H. -Bbuild \
```

File: recipes/recipe/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    - libblas * *openblas  # [linux]
+    - openblas-ilp64  # [linux]
+    #- libblas * *mkl  # [linux]
+    #- mkl  # [linux]
+    - liblapack
+    - python
+    - pandas >=0.14.1
+    - statsmodels >=0.11.0
+    - hdf5
+    - libuuid
```
Comment (xhochy) [inline recipes/recipe/meta.yaml:62]: This has a run_exports and will be added automatically here
```suggestion
```

File: recipes/hande/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    - scalapack
+    - python
+    - pandas >=0.14.1
+    - statsmodels >=0.11.0
+    - hdf5
+
+test:
+  commands:
+    - test -f ${PREFIX}/bin/hande.cmake.x
+    - test -f ${PREFIX}/lib64/libhande.a
```
Comment (xhochy) [inline recipes/hande/meta.yaml:47]: Can you please build the shared library here instead?

Comment (ltalirz) [reply recipes/hande/meta.yaml:47]: Thanks, I'd be happy to but I'm not sure there is a way to do that?
At least in the docs I only see the static one https://hande.readthedocs.io/en/latest/manual/compile-with-cmake.html#installation

File: recipes/hande/build.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+set -ex
+
+
+if [[ "$mpi" == "nompi" ]]; then
+
+cmake ${CMAKE_ARGS} -H. -Bbuild \
+    -DCMAKE_INSTALL_PREFIX=${PREFIX} \
```
Comment (xhochy) [inline recipes/hande/build.sh:8]: This should do the trick (general CMake feature)
```suggestion
    -DCMAKE_INSTALL_PREFIX=${PREFIX} \
    -DBUILD_SHARED_LIBS=ON
```

Comment (ltalirz) [reply recipes/hande/build.sh:8]: Cheers!

File: recipes/hande/meta.yaml (hunk 6 above)
Comment (xhochy) [reply recipes/hande/meta.yaml:47]: This test still passes, that is not a good sign :(

Comment (ltalirz) [reply recipes/hande/meta.yaml:47]: Yeah, it seems like ` cmake . --target install`  still only installs that library for static linking ([logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=453964&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1649)):
``` 
Install the project...
-- Install configuration: "Release"
-- Installing: $PREFIX/lib64/libhande.a
-- Installing: $PREFIX/bin/hande.cmake.x
```

**Comment by ltalirz** [comment]
@carterbox I've followed the suggestion but it seems this code does not want to create dynamically linked build.

If there's anything left to do on my side, just let me know

**Review by carterbox — CHANGES_REQUESTED** [review]
1. libhande is not built as a shared library because [they hardcode this library as a static library](https://github.com/hande-qmc/hande/blob/main/src/CMakeLists.txt), but our packaging policy is that all libraries should be offered as a shared library. So that's why CMAKE is ignoring the request for shared libraries. You will need to apply a patch, so the library that is generated as shared. Remember to add a run_export after switching to shared library.

2. The hande source code contains parts extracted from external libraries with separate licenses. These parts are then statically linked into libhande. These licenses need to be included in the license list so conda can extract them and include them with our distribution.

4. pyhande. Your recipe depends on python, so I assume you are also shipping pyhande, but you haven't tested this component.

5. pyhande and libhande should be separate outputs or recipes, so c/c++ consumers of libhande are not forced to install python.

**Pull request closed without merging**